### PR TITLE
llvm7: split libclang into own package

### PR DESCRIPTION
**Pull request by purpurrot**

### CODE REVIEW

**Comment by maxice8** [comment]
Please add [ci skip] the end of all commits in this Pull Request.

This Pull request takes over 50 minutes to package per-job on
Travis CI and is automatically terminated with a failure.

File: srcpkgs/llvm7/template [hunk 1]
```diff
@@ -190,24 +190,48 @@ clang-tools-extra_package() {
...
+libclang-devel_package() {
 	lib32disabled=yes
-	depends="libstdc++-devel libgcc-devel binutils"
+	depends="libstdc++-devel libgcc-devel binutils libclang"
 	case "$XBPS_TARGET_MACHINE" in
 		*-musl) depends+=" musl-devel";;
 		*) depends+=" glibc-devel";;
 	esac
-	short_desc+=" - C language family frontend"
+	short_desc+=" - Clang library development files "
```
Comment (maxice8) [inline srcpkgs/llvm7/template]: extra whitespace at the end

File: srcpkgs/llvm7/template [hunk 2]
```diff
@@ -190,24 +190,48 @@ clang-tools-extra_package() {
...
+clang_package() {
+	lib32disabled=yes
+	depends="libstdc++-devel libgcc-devel binutils libclang-devel"
+	short_desc+=" - C language family frontend"
+	alternatives="cc:cc:/usr/bin/clang"
+	homepage="https://clang.llvm.org/"
+	pkg_install() {
+		vmove "usr/bin/*clang*"
+		vmove usr/bin/c-index-test
 		vmove "usr/share/clang"
```
Comment (maxice8) [inline srcpkgs/llvm7/template]: doesn't need to be quoted since it has no wildcards.

**Comment by maxice8** [comment]
Needs to bump revision so the package gets rebuilt, please add [ci skip]

**Comment by Duncaen** [comment]
Any reason to do this?

File: srcpkgs/llvm7/template [hunk 3]
```diff
@@ -190,25 +190,49 @@ clang-tools-extra_package() {
 	}
 }
 
-clang_package() {
+libclang_package() {
+	lib32disabled=yes
+	depends="libstdc++ libgcc binutils"
```
Comment (Duncaen) [inline srcpkgs/llvm7/template]: It shouldn't be necessary to list those dependencies, except `binutils`.

File: srcpkgs/llvm7/template [hunk 4]
```diff
@@ -190,25 +190,49 @@ clang-tools-extra_package() {
...
+	case "$XBPS_TARGET_MACHINE" in
+		*-musl) depends+=" musl";;
+		*) depends+=" glibc";;
+	esac
```
Comment (Duncaen) [inline srcpkgs/llvm7/template]: This should also go.

File: srcpkgs/llvm7/template [hunk 5]
```diff
@@ -190,25 +190,49 @@ clang-tools-extra_package() {
...
+	homepage="https://clang.llvm.org/"
+	pkg_install() {
+		vmove "usr/lib/libclang.so*"
+	}
+}
+
+libclang-devel_package() {
 	lib32disabled=yes
-	depends="libstdc++-devel libgcc-devel binutils"
+	depends="libstdc++-devel libgcc-devel binutils libclang"
```
Comment (Duncaen) [inline srcpkgs/llvm7/template]: this should depend on the exact version of `libclang`.

**Comment by maxice8** [comment]
It has been discussed in IRC that it is best to do this change when updating LLVM to version 8

**Pull request closed without merging**